--- deltarpm-3.4/Makefile.rpm5	2010-12-05 19:26:34.979400433 +0100
+++ deltarpm-3.4/Makefile	2010-12-05 19:27:06.906157731 +0100
@@ -5,29 +5,34 @@ mandir=$(datadir)/man
 rpmdumpheader=$(bindir)/rpmdumpheader
 zlibdir=zlib-1.2.2.f-rsyncable
 CFLAGS = -O2 -Wall
-CPPFLAGS = -DDELTARPM_64BIT -DBSDIFF_NO_SUF -DRPMDUMPHEADER=\"$(rpmdumpheader)\" -I$(zlibdir)
-LDLIBS = -lbz2 $(zlibdir)/libz.a
-LDFLAGS =
+CPPFLAGS = -DDELTARPM_64BIT -DBSDIFF_NO_SUF -DRPMDUMPHEADER=\"$(rpmdumpheader)\" -I$(zlibdir) $(shell pkg-config --cflags rpm)
+LIBS = -lbz2 $(zlibdir)/libz.a $(shell pkg-config --libs rpm)
 
 all: makedeltarpm applydeltarpm rpmdumpheader makedeltaiso applydeltaiso combinedeltarpm fragiso
 
 makedeltarpm: makedeltarpm.o writedeltarpm.o md5.o util.o rpml.o rpmhead.o cpio.o delta.o cfile.o $(zlibdir)/libz.a
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) $^ $ -o $@ $(LIBS)
 
 applydeltarpm: applydeltarpm.o readdeltarpm.o md5.o util.o rpmhead.o cpio.o cfile.o prelink.o $(zlibdir)/libz.a
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) $^ $ -o $@ $(LIBS)
 
 combinedeltarpm: combinedeltarpm.o md5.o util.o rpmhead.o cfile.o readdeltarpm.o writedeltarpm.o $(zlibdir)/libz.a
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) $^ $ -o $@ $(LIBS)
 
 rpmdumpheader: rpmdumpheader.o
-	$(CC) $(LDFLAGS) $^ -lrpm -o $@
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) $^ $ -o $@ $(LIBS)
 
 makedeltaiso: makedeltaiso.o delta.o rpmoffs.o util.o md5.o cfile.o $(zlibdir)/libz.a
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) $^ $ -o $@ $(LIBS)
 
 applydeltaiso: applydeltaiso.o util.o md5.o cfile.o $(zlibdir)/libz.a
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) $^ $ -o $@ $(LIBS)
 
 fragiso: fragiso.o util.o md5.o rpmhead.o cfile.o $(zlibdir)/libz.a
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) $^ $ -o $@ $(LIBS)
 
 $(zlibdir)/libz.a:
-	cd $(zlibdir) ; make CFLAGS="$(CFLAGS)" libz.a
+	cd $(zlibdir) ; make CPPFLAGS="$(CPPFLAGS)" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" libz.a
 
 clean:
 	rm -f *.o
@@ -39,7 +44,7 @@ install:
 	install -m 755 makedeltarpm  $(DESTDIR)$(bindir)
 	install -m 755 applydeltarpm $(DESTDIR)$(bindir)
 	install -m 755 combinedeltarpm $(DESTDIR)$(bindir)
-	install -m 755 rpmdumpheader $(DESTDIR)$(rpmdumpheader)
+	install -m 755 rpmdumpheader -D $(DESTDIR)$(rpmdumpheader)
 	install -m 755 makedeltaiso $(DESTDIR)$(bindir)
 	install -m 755 applydeltaiso $(DESTDIR)$(bindir)
 	install -m 755 fragiso $(DESTDIR)$(bindir)
--- deltarpm-3.4/rpmdumpheader.c.rpm5	2007-03-06 14:31:13.000000000 +0100
+++ deltarpm-3.4/rpmdumpheader.c	2010-12-05 19:26:34.981398757 +0100
@@ -7,15 +7,23 @@
 
 #include <fcntl.h>
 #include <string.h>
-#include <rpm/rpmlib.h>
-#include <rpm/rpmts.h>
-#include <rpm/rpmdb.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <rpmrc.h>
+#include <rpmtypes.h>
+#include <rpmtag.h>
+#include <rpmlog.h>
+#include <rpmio.h>
+#include <rpmts.h>
+#include <rpmdb.h>
+#include <pkgio.h>
 
 int main(int argc, char **argv)
 {
   FD_t fdo;
   rpmts ts = NULL;
-  rpmdbMatchIterator mi;
+  rpmmi mi;
   Header h;
   int ret = 0;
 
@@ -59,13 +67,21 @@ int main(int argc, char **argv)
   ts = rpmtsCreate();
   
   mi = rpmtsInitIterator(ts, RPMTAG_NAME, n, 0);
-  rpmdbSetIteratorRE(mi, RPMTAG_EPOCH, RPMMIRE_STRCMP, e);
-  rpmdbSetIteratorRE(mi, RPMTAG_VERSION, RPMMIRE_STRCMP, v);
-  rpmdbSetIteratorRE(mi, RPMTAG_RELEASE, RPMMIRE_STRCMP, r);
-  if ((h = rpmdbNextIterator(mi)) != NULL)
-    {
-      fdo = Fopen("-", "w.ufdio");
-      headerWrite(fdo, h, HEADER_MAGIC_YES);
+  rpmmiAddPattern(mi, RPMTAG_EPOCH, RPMMIRE_STRCMP, e);
+  rpmmiAddPattern(mi, RPMTAG_VERSION, RPMMIRE_STRCMP, v);
+  rpmmiAddPattern(mi, RPMTAG_RELEASE, RPMMIRE_STRCMP, r);
+  if ((h = rpmmiNext(mi)) != NULL)
+    {
+	rpmRC rc;
+	const char item[] = "Header";
+	const char * msg = NULL;
+	
+      	fdo = Fopen("-", "w.ufdio");
+	rc = rpmpkgWrite(item, (FD_t)fdo, h, &msg);
+	if (rc != RPMRC_OK) {
+		rpmlog(RPMLOG_ERR, "%s: %s: %s\n", "headerWrite", item, msg);
+	}
+	msg = (const char*)_free(msg);
     }
   else
     {
@@ -75,7 +91,7 @@ int main(int argc, char **argv)
 	fprintf(stderr, "%s-%s-%s is not installed\n", n, v, r);
       ret = 1;
     }
-  mi = rpmdbFreeIterator(mi);
+  mi = rpmmiFree(mi);
   ts = rpmtsFree(ts);
   exit(ret);
 }
